### ISSUE TITLE
Malformed DOIs for references

### ISSUE THREAD

**Issue by coffeejunk** [sharkipedia/sharkipedia#41]
@creeas I was importing the resources from `docs/shark-resources.190605.csv` and noticed that the following ones have malformed DOIs:

[failed.txt](https://github.com/coffeejunk/sharkT/files/3291900/failed.txt)

**Comment by creeas** [comment]
Hey @coffeejunk I have had a look through the list of refs that came back with malformed DOIs. While some of them look wonky (see example below), the DOIs are valid and I think the catch has to do with the regular expression we are using. Can you send the check you are using and I can see if I can find something that will include these. 

![Screen Shot 2019-06-18 at 10 49 34 AM](https://user-images.githubusercontent.com/4634830/59707250-193d4800-91b7-11e9-9d2b-e9ac6fb2d2d6.png)

Alternatively I can check and see if these refs are really needed for the traits database and remove those that aren't. Because the resources list contain every resource from shark-references.com, many of them will not be germane to the traits database...

**Comment by coffeejunk** [comment]
@creeas the regex we're using for validation is this one:

https://github.com/coffeejunk/sharkT/blob/4a3070b435a49451179095407b41f342976fc6b8/app/models/resource.rb#L8

Happy to swap this out if you have a better one?

**Comment by sebpardo** [comment]
I'll leave this here for reference, from #17:

doi regex: `^10.\d{4,9}/[-._;()/:A-Z0-9]+$` or `\b(10[.][0-9]{4,}(?:[.][0-9]+)*/(?:(?!["&\'<>])\S)+)\b` (see https://www.regextester.com/96937 and https://www.crossref.org/blog/dois-and-matching-regular-expressions/)

**Comment by sebpardo** [comment]
Just a thought, can you use multiple regexps to check whether any doi matches one or more?

**Comment by coffeejunk** [comment]
>  can you use multiple regexps
it's possible but.. I wouldn't encourage it.. That said I've updated the regex to this:
https://github.com/coffeejunk/sharkT/blob/3d20af84fa23c011a865f40dc01566185b687078/app/models/resource.rb#L9-L34

and now all resources but the following four validate:

```
"717","ari2016","ARI, C. & D’AGOSTINO, D.P. (2016) Contingency checking and self-directed behaviors in giant manta rays: Do elasmobranchs have self-awareness? Journal of Ethology, 34 (2): 167–174 ",2016,"10.​1007/​s10164-016-0462-z","Shark-references","","ari2016"
"6125","dias2014","DIAS, J.F. & FERNANDEZ, W.S. & SCHMIDT, T.C.S. (2014) Length-weight relationship of 73 fish species caught in the southeastern inner continental shelf region of Brazil. Latin American Journal of Aquatic Research, 42 (1): 127–136 ",2014,"103856/vol42-issue1-fulltext-10","Shark-references","","dias2014"
"16566","narvaez2017","NARVÁEZ, K. & OSAER, F. (2017) The marine leech Stibarobdella macrothela parasitic on the angel shark Squatina squatina. Marine Biodiversity, 47 (1): 153–154 ",2017,"10.​1007/​s12526-016-0444-y","Shark-references","","narvaez2017"
"24209","weijs2015","WEIJS, L. & BRIELS, N. & ADAMS, D.H. & LEPOINT, G. & DAS, K. & BLUST, R. & COVACI, A. (2015) Maternal transfer of organohalogenated compounds in sharks and stingrays. Marine Pollution Bulletin, 92 (1–2): 59–68 ",2015,"1016/j.marpolbul.2014.12.056","Shark-references","","weijs2015"
```

`ari2016` and `narvaez2017` had some weird unicode whitespace in them so that was easy to fix (5be9b02) and so this leaves us with:

```
"6125","dias2014","DIAS, J.F. & FERNANDEZ, W.S. & SCHMIDT, T.C.S. (2014) Length-weight relationship of 73 fish species caught in the southeastern inner continental shelf region of Brazil. Latin American Journal of Aquatic Research, 42 (1): 127–136 ",2014,"103856/vol42-issue1-fulltext-10","Shark-references","","dias2014"

"24209","weijs2015","WEIJS, L. & BRIELS, N. & ADAMS, D.H. & LEPOINT, G. & DAS, K. & BLUST, R. & COVACI, A. (2015) Maternal transfer of organohalogenated compounds in sharks and stingrays. Marine Pollution Bulletin, 92 (1–2): 59–68 ",2015,"1016/j.marpolbul.2014.12.056","Shark-references","","weijs2015"
```

@creeas / @sebpardo any thoughts?

**Comment by sebpardo** [comment]
These two DOIs are misspelled. All DOIs should begin with 10.* so the `dias2014` reference is missing the dot, it's doi should be: [10.3856/vol42-issue1-fulltext-10](http://doi.org/10.3856/vol42-issue1-fulltext-10) (journal website seems to be down but the DOI redirect works)

Same with `weijs2015`, it's missing the initial `10.`: [10.1016/j.marpolbul.2014.12.056](http://doi.org/10.1016/j.marpolbul.2014.12.056)
